### PR TITLE
Implement support to mint pkp upon WebAuthn registration

### PR DESCRIPTION
**Pull request by hwrdtm**
# What

LIT-67

This PR introduces a much safer improvement to the WebAuthn PKP registration flow. Specifically,

- Added contract ABIs for SoloNet PKPs (while SoloNet is not meant to ever be deployed beyond our next launch, it is very useful for testing against `develop` branch Lit Nodes.
- Changed `expectedOrigin` to receive a `string[]`.
- Added `GET /auth/webauthn/generate-registration-options` and `POST /auth/webauthn/verify-registration` for apps to register against our relay.
  - We use the client-side username to derive the `authMethodId` for the minted PKP.
  - If a PKP already exists against a username, no minting occurs.
  - We don't work with challenges in the verify registration endpoint.
  - We store the entire hex-encoded COSE key on-chain for successful recovery of key material anywhere else, ie. on the Lit Node.
  - Use rpID as domain from request header Origin - this allows us to have other domains / projects act as an RP to prompt users to do WebAuthn.
- `mintPKP` method now supports minting against either SoloNetPKP or PKPNFT contracts.
- Use Redis (ZADD, ZRANGEBYSCORE) to maintain a list of PKP public keys for use when minting against SoloNet.
- Added a script to load PKP public keys (after they have been DKG'ed) into the Redis server using ZADD, all with score 0 (marked as "unminted"). This is only needed for usage against the SoloNet.

# Testing

- Use OAuth PKP repo to test against this. Branch name TBD.

### CODE REVIEW

File: config/index.ts [hunk 1]
```diff
@@ -2,11 +2,40 @@ require("dotenv").config();
...
+	pkpHelperAddress:
+		PKP_HELPER_ADDRESS || "0xffD53EeAD24a54CA7189596eF1aa3f1369753611",
+	pkpPermissionsAddress:
+		PKP_PERMISSIONS_ADDRESS || "0x274d0C69fCfC40f71E57f81E8eA5Bd786a96B832",
 };
 
-export default config;
+export default {
+	...baseConfig,
+	origin: baseConfig.enableHttps
```
Comment (hwrdtm) [inline config/index.ts:38]:
```suggestion
	expectedOrigin: baseConfig.enableHttps
```

File: config/index.ts [hunk 2]
```diff
@@ -2,11 +2,40 @@ require("dotenv").config();
...
+		? `https://${baseConfig.rpID}`
+		: `http://localhost:3000`,
```
Comment (hwrdtm) [inline config/index.ts:40]:
```suggestion
		: `http://localhost:3001`,
```
this can't collide with the same port that the relay server is running locally.

**Review by hwrdtm — COMMENTED** [review]
notes to self

File: public/index.html [hunk 3]
```diff
@@ -191,12 +191,15 @@ <h1>SimpleWebAuthn Example Site</h1>
...
-          const verificationResp = await fetch('/verify-registration', {
+          const verificationResp = await fetch('/auth/webauthn/verify-registration', {
             method: 'POST',
             headers: {
               'Content-Type': 'application/json',
             },
-            body: JSON.stringify(attResp),
+            body: JSON.stringify({
+              credential: attResp,
+              username: "test-user"
```
Comment (joshLong145) [inline public/index.html:201]: should this be hard coded?

**Review by joshLong145 — APPROVED** [review]

File: routes/auth/webAuthn.ts [hunk 4]
```diff
@@ -1,75 +1,182 @@
...
+	// Get username from query string
+	const username = req.query.username as string | undefined;
+
+	// Get RP_ID from request Origin.
+	const rpID = getDomainFromUrl(req.get("Origin") || "localhost");
+
+	const opts: GenerateRegistrationOptionsOpts = {
+		rpName: "Lit Protocol",
+		rpID,
+		userID: generateUsernameForOptions(username),
```
Comment (sarahzdev) [inline routes/auth/webAuthn.ts:42]: The [spec](https://w3c.github.io/webauthn/#user-handle) recommends that the `userId` be random bytes. Might be worth revisiting in this or the next PR.

**Review by sarahzdev — APPROVED** [review]

File: public/index.html (hunk 3 above)
Comment (hwrdtm) [reply public/index.html]: Resolved and removed.

File: routes/auth/webAuthn.ts (hunk 4 above)
Comment (hwrdtm) [reply routes/auth/webAuthn.ts:42]: You mean in the case of usernameless, we use random bytes instead of `Usernameless....` ?

Comment (hwrdtm) [reply routes/auth/webAuthn.ts:42]: Maybe we do `!!username ? username : defaultUsername()` for `userName` field and we do `userID` as the hash of the `userName` field?

**Pull request merged**